x/posts/keeper/keeper.go

```diff
@@ -42,9 +42,26 @@ func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, paramSpace params.Subspa
 func (k Keeper) SavePost(ctx sdk.Context, post types.Post) {
 	store := ctx.KVStore(k.StoreKey)
 
+	// Retrieve the total number of posts, if null it will be equal to 0
+	var numberOfPosts sdk.Int
+	if bz := store.Get(types.PostTotalNumberPrefix); bz != nil {
+		k.Cdc.MustUnmarshalBinaryBare(bz, &numberOfPosts)
+	} else {
+		numberOfPosts = sdk.NewInt(0)
+	}
+
 	// Save the post
 	store.Set(types.PostStoreKey(post.PostID), k.Cdc.MustMarshalBinaryBare(&post))
 
+	// Check if the postID got an associated index, if so, don't increment the number of posts
+	if store.Get(types.PostIndexedIDStoreKey(post.PostID)) == nil {
+		numberOfPosts = numberOfPosts.Add(sdk.NewInt(1))
```

Comment:
Since `numberOfPosts` will be used only inside this `if`, I think it's better to initialize it here to avoid any unnecessary initialization otherwise.

Reply:
I think we could move the whole `numberOfPosts` check inside the `if` statement so that we check on the store only if a post has never been added before.

Reply:
@bragaz Yes, that would be great!